llvm/lib/Target/RISCV/RISCVInstrInfo.cpp

```diff
@@ -4232,6 +4232,8 @@ unsigned RISCV::getDestLog2EEW(const MCInstrDesc &Desc, unsigned Log2SEW) {
 
 /// Given two VL operands, do we know that LHS <= RHS?
 bool RISCV::isVLKnownLE(const MachineOperand &LHS, const MachineOperand &RHS) {
+  if (LHS.isImm() && LHS.getImm() == 0)
+    return true;
```

Comment:
This is needed to be able to replace the std::optional<MachineOperand> with CreateImm(0) in computeDemandedVL